README.md

```diff
@@ -130,13 +137,50 @@ posts for further information
     </td><td><code>""</code></td></tr>
 
   <tr><td>esHeapSize</td><td>integer</td>
-    <td>The size, in megabytes, of memory to allocate on each Elasticsearch node for the JVM heap. If unspecified, 50% of the available memory will be allocated to Elasticsearch heap, up to a maximum of 31744MB (~32GB). 
-    Take a look at <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/heap-size.html" target="_blank">the Elasticsearch documentation</a> for more information.  <strong>This is an expert level feature - setting a heap size larger than available memory on the Elasticsearch VM SKU will fail the deployment.</strong>
+    <td>The size, <em>in megabytes</em>, of memory to allocate on each Elasticsearch node for the JVM heap. If unspecified, 50% of the available memory will be allocated to Elasticsearch heap, up to a maximum of 31744MB (~32GB). 
+    Take a look at <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/heap-size.html" target="_blank">the Elasticsearch documentation</a> for more information.  <strong>This is an expert level feature - setting a heap size too low, or larger than available memory on the Elasticsearch VM SKU will fail the deployment.</strong>
     </td><td><code>0</code></td></tr>
 
+  <tr><td>esHttpSecurity</td><td>string</td>
+    <td>Either <code>Yes</code> or <code>No</code> to configure SSL/TLS for the HTTP layer of Elasticsearch. <strong>X-Pack plugin must be installed and either <code>esHttpCertBlob</code> or <code>esHttpCaCertBlob</code> provided</strong></strong>
+    </td><td><code>No</code></td></tr>
+
+  <tr><td>esHttpCertBlob</td><td>string</td>
+    <td>A Base-64 encoded form of the PKCS#12 archive (.p12/.pfx) certificate to secure communication for HTTP layer to Elasticsearch. <strong>X-Pack plugin must be installed</strong>
```

Comment:
I assume you don't want to pass the PKCS#12 Keystores as Base64 encoded strings but rather pass the file directly as a param right ? This goes for the rest of the settings below.
Rephrase to something like: _A PKCS#12 archive (.p12/.pfx) containing the key and certificate needed to secure communication for the HTTP layer of Elasticsearch_ ?

Reply:
The PKCS#12 archives need to be base 64 encoded as all parameters are passed over HTTP to the template, whether the template is part of a Marketplace offering or hosted publicly. 

If we end up exposing the ability for users to upload certificate files in [the Azure Marketplace UI](https://azuremarketplace.microsoft.com/en-au/marketplace/apps/elastic.elasticsearch?tab=Overview), The file upload control will base 64 encode content to pass it to the template

Reply:
Just re-read this comment. I'll update the description to make it more clear.

Reply:
> we're asking users to take a PKCS#12 archive, find a way to represent its bytes in Base64 format and use that string here. I'm not familiar with how Azure templates work but I assumed there is an easy way to add the actual binary file here instead of its Base64 string representation.

Yes, that's what we're asking users to do, and it's the way the contents are files are typically provided when deploying Azure resources.

> Also I still think that the PKCS#12 archive certificate part sounds strange as there are two nouns. If the containing the key and certificate I suggested above sounds too long, maybe consider dropping the certificate from the original description to keep it simple ?

I agree, I'll update to your suggestion _"containing the key and certificate"_ 👍

Reply:
> I agree, I'll update to your suggestion "containing the key and certificate" +1

I'd update that here (in the Parameters table) too.